pkg/controllers/deprovisioning/validation.go

```diff
@@ -97,6 +101,30 @@ func (v *Validation) IsValid(ctx context.Context, cmd Command) (bool, error) {
 	return isValid, nil
 }
 
+// Returns if the candidates chosen can still be terminated after the TTL
+func (v *Validation) validateCandidates(cmd Command, pdbs *PDBLimits) bool {
+	for _, n := range cmd.candidates {
+		if !n.Node.DeletionTimestamp.IsZero() {
+			v.recorder.Publish(deprovisioningevents.Unconsolidatable(n.Node, "in the process of deletion")...)
+			return false
+		}
+		if p, ok := hasDoNotEvictPod(n); ok {
+			v.recorder.Publish(deprovisioningevents.Unconsolidatable(n.Node, fmt.Sprintf("pod %s/%s has do not evict annotation", p.Namespace, p.Name))...)
+			return false
+		}
+		if pdb, ok := pdbs.CanEvictPods(n.pods); !ok {
+			v.recorder.Publish(deprovisioningevents.Unconsolidatable(n.Node, fmt.Sprintf("pdb %s prevents pod evictions", pdb))...)
+			return false
+		}
+		// a candidate we are about to delete is a target of a currently pending pod, wait for that to settle
+		// before continuing consolidation
+		if v.cluster.IsNodeNominated(n.Name()) {
```

Comment:
Should we check this first, to shortcircuit?

Reply:
I can change this to be checked first. 

On the second point: 
We do filter out nodes that are terminating, nominated, or not initialized already when we compute candidates. Rather than re-computing all the candidate nodes of the cluster, it's cheaper to check after some time if these existing candidate nodes for this deprovisioning command are either 1) still terminating, 2) now has a do-not-evict pod 3) now has a blocking PDB. Thoughts?

Reply:
Is there a method we can factor out that checks whether or not a node is viable for either the first pass, or the second pass of consolidation? I want to make sure we're not replicating logic and potentially introducing bugs (now, or in the future).

Reply:
We previously factored out a function to generally return if a node has blocking pods, but wanted to pull it out to take different actions for each condition. This is replicated as a way to optimize computing consolidation actions, but I think refactoring that part of the code is out of the scope of this PR.